owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/sharing/OCShareViewModel.kt

```diff
@@ -0,0 +1,366 @@
+    private val getShareAsLiveDataUseCase: GetShareAsLiveDataUseCase = GetShareAsLiveDataUseCase(context, account),
+    private val refreshSharesFromNetworkUseCase: RefreshSharesFromNetworkUseCase =
+        RefreshSharesFromNetworkUseCase(context, account),
+    private val createPrivateShareUseCase: CreatePrivateShareUseCase = CreatePrivateShareUseCase(context, account),
+    private val editPrivateShareUseCase: EditPrivateShareUseCase = EditPrivateShareUseCase(context, account),
+    private val createPublicShareUseCase: CreatePublicShareUseCase = CreatePublicShareUseCase(context, account),
+    private val editPublicShareUseCase: EditPublicShareUseCase = EditPublicShareUseCase(context, account),
+    private val deletePublicShareUseCase: DeleteShareUseCase = DeleteShareUseCase(context, account)
+) : ViewModel() {
+
+    private val _shares = MutableLiveData<UIResult<List<OCShareEntity>>>()
+    val shares: LiveData<UIResult<List<OCShareEntity>>> = _shares
+
+    private var sharesLiveData: LiveData<List<OCShareEntity>>? = getSharesAsLiveDataUseCase.execute(
+        GetSharesAsLiveDataUseCase.Params(
+            filePath = filePath,
+            accountName = account.name
+        )
+    ).data
+
+    // To detect changes in shares
+    private val sharesObserver: Observer<List<OCShareEntity>> = Observer { shares ->
+        _shares.postValue(UIResult.success(shares))
+    }
+
+    init {
+        sharesLiveData?.observeForever(sharesObserver)
+
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                refreshSharesFromNetwork()
+            }
+        }
+    }
+
+    private fun refreshSharesFromNetwork() {
+        _shares.postValue(
+            UIResult.loading(sharesLiveData?.value)
+        )
+
+        refreshSharesFromNetworkUseCase.execute(
+            RefreshSharesFromNetworkUseCase.Params(
+                filePath = filePath,
+                accountName = account.name
+            )
+        ).also { useCaseResult ->
+            if (!useCaseResult.isSuccess()) {
+                _shares.postValue(
+                    UIResult.error(
+                        sharesLiveData?.value,
+                        errorMessage = useCaseResult.msg ?: ErrorMessageAdapter.getResultMessage(
+                            useCaseResult.code,
+                            useCaseResult.exception,
+                            OperationType.GET_SHARES,
+                            context.resources
+                        )
+                    )
+                )
+            }
+        }
+    }
+
+    private val _shareDeletionStatus = MutableLiveData<UIResult<Unit>>()
+    val shareDeletionStatus: LiveData<UIResult<Unit>> = _shareDeletionStatus
+
+    fun deleteShare(
+        remoteId: Long
+    ) {
+        _shareDeletionStatus.postValue(
+            UIResult.loading()
+        )
+
+        viewModelScope.launch {
```

Comment:
the same here.

Well thinking better, review all files because I don't want to type always the same hehe